lib/rubygems/executor.rb

```diff
@@ -0,0 +1,70 @@
+# frozen_string_literal: true
+require 'open-uri'
+require 'json'
+require 'rubygems/platform'
+
+module Gem
+  module Web
+    class Executor
+
+      def open_page(gem, options)
+        begin
+          spec = Gem::Specification.find_by_name(gem)
+        rescue Gem::MissingSpecError => e
+          spec = fetch_remote_spec(gem)
+
+          if spec.nil?
+            puts "Could not find '#{gem}' in rubygems.org."
+            return
+          end
+        end
+
+        if options[:sourcecode]
+          get_info_from_metadata(spec, "source_code_uri")
+        elsif options[:doc]
+          get_info_from_metadata(spec, "documentation_uri")
+        elsif options[:rubygems]
+          open_rubygems(gem)
+        else # The default option is homepage
+          open_browser(spec.homepage)
+        end
+      end
+
+      def fetch_remote_spec(gem)
+        dep = Gem::Dependency.new gem
+        found, _ = Gem::SpecFetcher.fetcher.spec_for_dependency(dep)
+        spec_tuple = found.first
+
+        if spec_tuple.nil? || spec_tuple.empty?
+          nil
+        else
+          spec_tuple.first
+        end
```

Comment:
You can rewrite it like:
```ruby
spec_tuple.first unless spec_tuple.nil? || spec_tuple.empty?
```

Reply:
It would be `spec_tuple.first.strip unless spec_tuple.nil? || spec_tuple.empty?`.

Maybe `spec_tuple.first&.strip unless spec_tuple.nil? || spec_tuple.empty?`, but I don't know if there's a way to spec_tuple.first to be `nil`